File: apps/mobile-react-native/android/app/src/main/java/com/semitrax/nativebridge/GuidanceBoundary.kt
package com.semitrax.nativebridge
/** SDK-neutral boundary. Replace only after reviewing the official vendor SDK contract. */
interface GuidanceBoundary {
  fun command(command: String, payload: String): String
}
class UnconfiguredGuidanceBoundary : GuidanceBoundary {
  override fun command(command: String, payload: String): String {
    if (command == "stopNavigation") return """{"phase":"unavailable"}"""
    throw IllegalStateException("NATIVE_TRUCK_GUIDANCE_NOT_CONFIGURED")
  }
}

File: apps/mobile-react-native/ios/SemiTrax/GuidanceBoundary.swift
import Foundation
protocol GuidanceBoundary {
  func command(_ command: String, payload: String) throws -> String
}
enum GuidanceFailure: Error { case notConfigured }
final class UnconfiguredGuidanceBoundary: GuidanceBoundary {
  func command(_ command: String, payload: String) throws -> String {
    if command == "stopNavigation" { return "{\"phase\":\"unavailable\"}" }
    throw GuidanceFailure.notConfigured
  }
}

